evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -155,35 +161,45 @@ def transform(self, X, y=None):
         if y is not None:
             y = infer_feature_types(y)
 
+        # This will change the logical type of BooleanNullable/IntegerNullable/AgeNullable columns with nans
+        # so we save the original schema to recreate it where possible after imputation
+        original_schema = X.ww.schema
+        X, y = self._handle_nullable_types(X, y)
+
         X_not_all_null = X.ww.drop(self._all_null_cols)
-        X_schema = X_not_all_null.ww.schema
-        X_schema = X_schema.get_subset_schema(
-            subset_cols=X_schema._filter_cols(
-                exclude=["IntegerNullable", "BooleanNullable", "AgeNullable"],
-            ),
+        original_schema = original_schema.get_subset_schema(
+            subset_cols=list(X_not_all_null.columns),
         )
+        new_ltypes = None
 
         if self._forwards_cols is not None:
-            X_forward = X.ww[self._forwards_cols]
+            X_forward = X[self._forwards_cols]
             imputed = X_forward.pad()
             imputed.bfill(inplace=True)  # Fill in the first value, if missing
             X_not_all_null[X_forward.columns] = imputed
 
         if self._backwards_cols is not None:
-            X_backward = X.ww[self._backwards_cols]
+            X_backward = X[self._backwards_cols]
             imputed = X_backward.bfill()
             imputed.pad(inplace=True)  # Fill in the last value, if missing
             X_not_all_null[X_backward.columns] = imputed
 
         if self._interpolate_cols is not None:
-            X_interpolate = X.ww[self._interpolate_cols]
-            # TODO: Revert when pandas introduces Float64 dtype
-            imputed = X_interpolate.astype(
-                float,
-            ).interpolate()  # Cast to float because Int64 not handled
+            X_interpolate = X_not_all_null[self._interpolate_cols]
+            imputed = X_interpolate.interpolate()
             imputed.bfill(inplace=True)  # Fill in the first value, if missing
             X_not_all_null[X_interpolate.columns] = imputed
-        X_not_all_null.ww.init(schema=X_schema)
+
+            # Interpolate may add floating point values to integer data,
+            # so we'll want to use the Double logical type
+            # --> this will lose the age ltype and I'm okay with that in this edge case
+            new_ltypes = {
+                col: Double
+                if isinstance(ltype, (IntegerNullable, AgeNullable))
```

Comment:
This will lose the AgeNullable logical type info, turning it into Double, but we doon't use the Age logical types a whole lot, and this is one small case where the extra logic didn't feel worth it. We can always add it in if we need it.